backend/app/services/api/filterer.rb

```diff
@@ -23,7 +25,8 @@ def call
       apply_numerical_filter
       filter_by_enums
       filter_by_only_verified
-      query
+      filter_by_relations
+      original_query.where id: query.pluck(:id)
```

Comment:
I have decided to use `pluck` here, because query generated by all filters (especially when multiple filters are combined) is getting super complex. This breaks this complexity so all additional stuff inside API (sorting, preloading, etc.) is using simple query instead of super complex one.